dvc/stage.py

```diff
@@ -262,18 +262,19 @@ def changed(self):
     def remove_outs(self, ignore_remove=False):
         """Used mainly for `dvc remove --outs` and :func:`Stage.reproduce`."""
         for out in self.outs:
-            logger.debug(
-                "Removing output '{out}' of '{stage}'.".format(
-                    out=out, stage=self.relpath
+            if out.persist:
+                out.unprotect()
+            else:
+                logger.debug(
+                    "Removing output '{out}' of '{stage}'.".format(
+                        out=out, stage=self.relpath
+                    )
                 )
-            )
-            out.remove(ignore_remove=ignore_remove)
+                out.remove(ignore_remove=ignore_remove)
 
     def unprotect_outs(self):
         for out in self.outs:
-            if out.scheme != "local" or not out.exists:
```

Comment:
Previously we were simply ignoring an output if it didn't exist, but now `unprotect()` will raise an exception. This  will break `dvc repro` when output doesn't exist.

Reply:
This functionality still exists, I moved it to output base class.

Reply:
Ah, indeed! Sorry, didn't catch that :slightly_smiling_face: